index_document.go

```diff
@@ -284,7 +284,7 @@ func (i *index) GetDocumentsWithContext(ctx context.Context, param *DocumentsQue
 		acceptedStatusCodes: []int{http.StatusOK},
 		functionName:        "GetDocuments",
 	}
-	if param != nil && param.Filter == nil {
+	if param != nil && param.Filter == nil && len(param.Ids) == 0 && len(param.Sort) == 0 {
```

Comment:
Getting documents with the GET method is deprecated and will be removed soon. Instead of adding complexity here, I suggest we only implement the POST `/fetch` method.

https://www.meilisearch.com/docs/reference/api/documents#get-documents-with-get

Please make simpler here.